lighthouse-core/report/html/renderer/performance-category-renderer.js

```diff
@@ -117,7 +117,8 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
     if (fmp) v5andv6metrics.push(fmp);
 
     const metricPairs = v5andv6metrics.map(audit => {
-      const value = audit.result.numericValue ? audit.result.numericValue.toString() : 'null';
+      const value = typeof audit.result.numericValue !== 'undefined' ?
```

Comment:
wdyt of just


```
const value = audit.result.numbericValue === undefined ? null : audit.result.numericValue
```

?

I don't think you need to coerce to string (params will handle that) or use typeof (just check if it is undefined)

Reply:
> I don't think you need to use typeof

old habits die hard my friend, I don't think I'm ever going to remember to *not* use typeof for undefined at this point 😆

Reply:
you only need it when you are trying to check if a value in the global scope is defined.

```
if (window === undefined) // whoops this throws an error when there is no window

if (typeof window === 'undefined') // fine, becausejs


let obj = {};
if (obj.blah === undefined) // fine
```

Reply:
i'm sure you know this already but it took me a couple years :)

Reply:
Consistency is definitely a good reason, but not the only reason 😉 

Once upon a time `undefined` was not reserved so you could do `undefined = 'hello'`. In the heyday of WordPress development and crazy jQuery plugins, I worked on more than one project where someone decided it was a good idea to set `undefined = null` so that they could do easier comparisons. If you wrote code for a client that could potentially live with such crazy other global nonsense and *really* wanted to test for undefinedness, you either had to do `typeof o.blah === 'undefined'` or wrap your junk in an IIFE that accepts an argument called `undefined` so at least locally it was safe. 

e.g.

```js
(function (undefined) {
  // now undefined is safe again :)
})();
```

AFAIK, it hasn't *really* been a problem for the past decade or so since it was fixed in [ECMAScript 5.1](http://www.ecma-international.org/ecma-262/5.1/#sec-15.1.1.3), but like I said, old habits die hard 😄

Reply:
wow I would riot if you ppl touched undefined how did you constrain yourself.

huh I didn't realize it's technically a property defined on the global object

type `window.undefined` in CDT console and before you even hit enter you see the result is `undefined` (do some non-existent property and there is no preview)

altho i can't find it via Object.getProperties/whaterver